code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -92,24 +55,10 @@ public void run() {
                         .and(SERVICE_EXPOSE_MAP.SERVICE_ID.eq(serviceId))
                         .and(SERVICE_EXPOSE_MAP.STATE.in(CommonStatesConstants.ACTIVATING,
                                 CommonStatesConstants.ACTIVE, CommonStatesConstants.REQUESTED)))
```

Comment:
Sorry, you're right.  I was thinking about the from the API only.  In the API layer it is all one transaction, but you are absolutely right.  On the backend it isn't.  Good job, just testing you there :)